### PR TITLE
Token management transactions outcome parsers

### PR DESCRIPTION
**Pull request by popenta**
Added specifications for parsing the outcome of Token Management transactions.

The SC Results are not needed for parsing all transactions, only for some. For consistency reasons, i pass an object of type `ITransactionResultsAndLogsHolder` to all methods.

The `Logs` and `SC Results` should be the same as they come from the API when calling `api.multiversx.com/transactions/txHash`.

### CODE REVIEW

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## Interfaces needed for parsing token management transactions
+
+```
+dto ITransactionEvent:
+    address: string;
+    identifier: string;
+    topics: List[string]
+    data: string
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:8]: Perhaps as `bytes`?

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    events: List[ITransactionEvent];
+```
+
+```
+dto ITransactionResult:
+    hash: string;
+    timestamp: int;
+    sender: string;
+    receiver: string;
+    data: string;
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:23]: As `bytes`?

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+## Interfaces needed for parsing token management transactions
+
+```
+dto ITransactionEvent:
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:4]: If `dto`, without the prefix `I` (below, as well).

File: sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md [hunk 4]
```diff
@@ -1 +1,115 @@
-// TBD: will be defined in a future PR.
+## TokenManagementTransactionsOutcomeParser
+
+A class that provides methods for extracting the output of token management transactions.
+i.e. Getting the ticker of a newly issued ESDT token.
+
+Each method should ensure that there is no `signalError` event;
+
+For methods that return multiple things, plain object can be used (JS) or simple DTOs(Py).
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:8]: For consistency, how about always return a DTO, even if only one piece of information is extracted? A bit more extensible without having breaking changes. E.g. if we decide that we'd also want to "extract" the minted amount in `parse_issue_fungible`, we only need to add an extra field on the returned DTO.

File: sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md [hunk 5]
```diff
@@ -1 +1,115 @@
...
+    parse_set_special_role(transaction_results_and_logs: ITransactionResultsAndLogsHolder): {
+        user_address: string;
+        token_identifier: string;
+        roles: List[string];
+    };
+
+    parse_nft_create(transaction_results_and_logs: ITransactionResultsAndLogsHolder): {
+        token_identifier: string;
+        nonce: string;
+        initial_quantity: string;
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:54]: Numerical - thus `uint64` for nonce and `Amount` for `initial_quantity`?

Comment (andreibancioiu) [reply sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:54]: Also below, for nonces and amounts (supply, balance, quantity etc.).

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    sender: string;
+    receiver: string;
+    data: string;
+    original_tx_hash: string;
+    minblock_hash: string;
+    logs: ITransactionLogs;
+```
+
+```
+dto ITransactionResultsAndLogsHolder:
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:30]: Can be called `TransactionOutcome`.

File: sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md [hunk 7]
```diff
@@ -1 +1,115 @@
...
+```
+dto RegisterAndSetAllRolesOutcome:
+    token_identifier: str
+    roles: List[str]
+```
+
+```
+class TokenManagementTransactionsOutcomeParser:
+    // returns the token identifier
+    parse_issue_fungible(transaction_results_and_logs: ITransactionResultsAndLogsHolder): string;
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:23]: From the name of the class, we can say that the input argument can be `transaction_outcome: TransactionOutcome`.

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+dto ITransactionResult:
+    hash: string;
+    timestamp: int;
+    sender: string;
+    receiver: string;
+    data: string;
+    original_tx_hash: string;
+    minblock_hash: string;
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:25]: Hopefully, we do no need the miniblock hash - maybe remove it, until we learn is necessary? Also, the `original_tx_hash`, for now?

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+```
+dto ITransactionLogs:
+    address: string;
+    events: List[ITransactionEvent];
+```
+
+```
+dto ITransactionResult:
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:18]: `dto SmartContractResult`?

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    receiver: string;
+    data: string;
+    original_tx_hash: string;
+    minblock_hash: string;
+    logs: ITransactionLogs;
+```
+
+```
+dto ITransactionResultsAndLogsHolder:
+    transaction_results: List[ITransactionResult]
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:31]: `smart_contract_results`.

File: sdk-core/transactions-outcome-parsers/interfaces.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    transaction_logs: ITransactionLogs
```
Comment (andreibancioiu) [inline sdk-core/transactions-outcome-parsers/interfaces.md:32]: One log with more events or more logs?

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 1 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:8]: changed to `bytes`

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 2 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:23]: changed to `bytes`

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 3 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:4]: removed

File: sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md (hunk 4 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:8]: Yes, that makes sense. Changed.

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 6 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:30]: Renamed

File: sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md (hunk 7 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/token_management_transactions_outcome_parser.md:23]: Renamed to `transaction_outcome: TransactionOutcome`

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 8 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:25]: Removed `miniblock_hash` and `original_tx_hash`.

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 9 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:18]: Renamed

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 10 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:31]: renamed

File: sdk-core/transactions-outcome-parsers/interfaces.md (hunk 11 above)
Comment (popenta) [reply sdk-core/transactions-outcome-parsers/interfaces.md:32]: One log with more events. Renamed to `transaction_log`.

**Review by andreibancioiu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**